### DISCUSSION TITLE
Capture Audio Output of Subprocess using Pipewire

### DISCUSSION THREAD

**Discussion by ghost** [Soundux/rohrkabel#7] [Q&A] [answered]
Hello!

Thanks for your effort for making this project. I have a question regarding the possibility of a. Capturing the Audio output of an application (PID is known, rights/privileges are not an issue) of a running application and b. providing said application with a custom microphone input.

If you could provide me with some pointers/documentation regarding this, this would be appreciated.

**Comment by Curve** [comment, marked as answer]
a. You'd have to consult the pipewire documentation for this, it's probably possible to capture the output of a node with it's API but I'm not sure, a naive approach would be to link the node to another node that records it's input
b. Please take a look at [venmic](https://github.com/Vencord/venmic), which implements this (in case you want a code example) - In short: you need to link the output of one node to the input of the other, this can be done using links, see rohrkabels pipewire::link_factory for easy creation
